Lesson1HomeTask/task3/main.cpp

```diff
@@ -0,0 +1,94 @@
+#include <iostream>
+#include <forward_list>
+#include <vector>
+
+template <typename T>
+void print_list (const std::forward_list<T>& f_list)
+{
+    for (const auto& el : f_list)
+        std::cout << el << " ";
+}
+
+template <typename T>
+void reverse_list_1 (std::forward_list<T>& f_list)
+{
+    std::forward_list<T> tmp_flist;
+    for (const auto& el : f_list)
+        tmp_flist.push_front (el);
+    f_list = tmp_flist;
+}
+
+
+template <typename T>
+void reverse_list_2 (std::forward_list<T>& f_list)
+{
+    std::vector<typename std::forward_list<T>::iterator> f_list_iterators;
```

Comment:
Consider using alias here :)

Reply:
ok)

Reply:
I think, in c++14 it is not necessary, am I right?